executor/adapter.go

```diff
@@ -534,21 +534,22 @@ func (a *ExecStmt) handlePessimisticDML(ctx context.Context, e Executor) error {
 		if !txn.Valid() {
 			return err
 		}
+		keys, err1 := txn.(pessimisticTxn).KeysNeedToLock()
+		if err1 != nil {
+			return err1
+		}
 		if err != nil {
 			// It is possible the DML has point get plan that locks the key.
 			e, err = a.handlePessimisticLockError(ctx, err)
 			if err != nil {
 				if ErrDeadlock.Equal(err) {
 					metrics.StatementDeadlockDetectDuration.Observe(time.Since(startPointGetLocking).Seconds())
 				}
+				txn.CleanupKeyExistErrInfo(keys)
```

Comment:
May it mistakenly clean the `PresumeKeyNotExistsError` added by a previous statement, so that this flag is missing at commit phase.

Reply:
For pessimistic transaction, every duplicate keys are checked in the pessimistic lock phase, not in commit phase.

Reply:
This flag may still be used in the commit phase like deciding if `Op_Insert` should be used for pessimistic transactions. We still need to make sure if it won't be cleared mistakenly.

Reply:
If so, then the key won't be put into the memBuffer?